lib/config/config-initializer.js

```diff
@@ -279,11 +306,74 @@ function getConfigForStyleGuide(guide) {
         throw new Error("You referenced an unsupported guide.");
     }
 
-    installModules(guides[guide]);
+    installModules(guides[guide], installESLint);
 
     return guides[guide];
 }
 
+/**
+ * Get the version of the local ESLint.
+ * @returns {string|null} The version. If the local ESLint was not found, returns null.
+ */
+function getLocalESLintVersion() {
+    try {
+        const resolver = new ModuleResolver();
+        const eslintPath = resolver.resolve("eslint", process.cwd());
+        const eslint = require(eslintPath);
+
+        return eslint.linter.version || null;
+    } catch (_err) {
+        return null;
+    }
+}
+
+/**
+ * Get the shareable config name of the chosen style guide.
+ * @param {Object} answers The answers object.
+ * @returns {string} The shareable config name.
+ */
+function getStyleGuideName(answers) {
+    if (answers.styleguide === "airbnb" && !answers.airbnbReact) {
+        return "airbnb-base";
+    }
+    return answers.styleguide;
+}
+
+/**
+ * Check the local ESLint version satisfies the required version of the chosen shareable config.
+ * @param {Object} answers The answers object.
+ * @returns {boolean} `true` if the local ESLint is not found or it satisfies the required version of the chosen shareable config.
+ */
+function checkLocalESLintVersion(answers) {
```

Comment:
To make the meaning of this method's boolean return value more apparent, what do you think of renaming this to `hasESLintVersionConflict` and inverting the return values? (This would allow you to drop the negation when calling this function on [line 452](https://github.com/eslint/eslint/pull/8911/files#diff-da527c6839c0e6e4d343b5c77fcdd27cR452).)